cpp/create_module/algorithm/create.cpp

```diff
@@ -1,8 +1,73 @@
 #include "create.hpp"
-#include "_mgp.hpp"
-#include "mg_procedure.h"
+
+#include <unordered_set>
 #include "mgp.hpp"
 
+namespace {
+void ThrowException(const mgp::Value &value) {
+  std::ostringstream oss;
+  oss << value.Type();
+  throw mgp::ValueException("Unsupported type for this operation, received type: " + oss.str());
+}
+
+void DeleteAndOutput(mgp::Relationship &relationship, const mgp::List &keys, const mgp::RecordFactory &record_factory) {
+  for (const auto &key : keys) {
+    relationship.RemoveProperty(std::string(key.ValueString()));
+  }
+
+  auto record = record_factory.NewRecord();
+  record.Insert(std::string(Create::kResultRemoveRelProperties).c_str(), relationship);
+}
+
+}  // namespace
+
+void Create::RemoveRelProperties(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+  mgp::memory = memory;
+  const auto arguments = mgp::List(args);
+  const auto record_factory = mgp::RecordFactory(result);
+
+  try {
+    mgp::Graph graph{memgraph_graph};
+    const auto keys{arguments[1].ValueList()};
+
+    std::unordered_set<int64_t> ids;
+
+    auto ParseValue = [&](const mgp::Value &value) {
+      if (value.IsRelationship()) {
+        auto relationship_copy = value.ValueRelationship();
+        DeleteAndOutput(relationship_copy, keys, record_factory);
+      } else if (value.IsInt()) {
+        ids.insert(value.ValueInt());
+      } else {
+        ThrowException(value);
+      }
+    };
+
+    if (!arguments[0].IsList()) {
+      ParseValue(arguments[0]);
```

Comment:
You can return here, right?

Reply:
No because ParseValue can insert ids into the unordered_set (if id is given). I could add a check
`if (arguments[0].IsInt()) return...`
but I don't think its needed because few lines later that is checked